### ISSUE TITLE
Azure File Copy with ##vso[task.setvariable] does not work when system.debug is false

### ISSUE THREAD

**Issue by compulim** [Microsoft/vso-agent-tasks#1535]
On a build/release with `system.debug` set to `false`, a build task will not be able to pass variable back to the build process and let downstream consume it. `##vso[task.setvariable]` is simple not working.

Tried it on stock Azure File Copy task in VSTS 1.98.0.

**Comment by ericsciple** [comment]
I can't repro this using the standard Command Line tasks.
Task 1:
  Tool: echo
  Arguments: ##vso[task.setvariable variable=myvar]myvalue
Task 2:
  Tool: echo
  Arguments: myvar=$(myvar)

I'm guessing you are trying to set the variable in a previous task, and then consume the variable in the Azure File Copy task, correct?

**Comment by compulim** [comment]
Upstream is Azure File Copy and it is trying to output variable of _Storage Container URI_ and _Storage Container SAS Token_.

The downstream is Azure Resource Group Deployment, which I got a template to run MSDeploy to deploy to an Azure Web App. We need to use Azure RG because we are on Azure RM and not Azure Classic. The current Azure WebDeploy task only works for Azure Classic. I see the AzureRM WebDeploy is currently under development.

Anyway, if I set `system.debug` to `false`, the Azure Resource Group Deployment will get nothing for both _Storage Container URI_ and _Storage Container SAS Token_. I confirmed this with a simple PowerShell I wrote to dump both.

**Comment by ericsciple** [comment]
Thank you. I've updated the title so someone with more context on that task will notice. Trying to track down a good assignee now...

**Comment by ericsciple** [comment]
Krishna, this appears to be a dupe of this issue: https://github.com/Microsoft/vso-agent-tasks/issues/1523

I see 1523 was closed three days ago. Was the hotfix rolled out everywhere? Did it regress with 98 deployment?

**Comment by ericsciple** [comment]
...sorry I see it was closed 3 *hours* ago. @compulim have you tried within the last three hours :)

**Comment by compulim** [comment]
Not last 3 hours but within last 15 hours. Will try it out now.

**Comment by compulim** [comment]
My build tasks list still show the old version 1.0.52. It was 1.0.51 two days ago.

```
id            : eb72cb01-a7e5-427b-a8a1-1b31ccac8a43
name          : AzureFileCopy
friendly name : Azure File Copy
visibility    : Build,Release
description   : Copy files to Azure blob or VM(s)
version       : 1.0.52
```

The patched should be 1.0.53. Need some time to wait for the rollout?

**Comment by compulim** [comment]
It seems the fix is working despite my build task list show 1.0.52. Closing the issue now. Thanks @ericsciple.
